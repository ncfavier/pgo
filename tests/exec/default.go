package main
import "fmt"

/* valeurs par dÃ©faut :
	 int    0
	 bool   false
	 string ""
	 *tau   nil
*/

type T struct { x int; y bool; z *int }

func main() {
	var b bool
	var i int
	var s string
	var t T
	var p *T
    var q = new(T)
	fmt.Print(b, "\n")
	fmt.Print(i, "\n")
	fmt.Print(s, "\n")
	fmt.Print(t.x, "\n")
	fmt.Print(t.y, "\n")
	fmt.Print(t.z, "\n")
	fmt.Print(p, "\n")
	fmt.Print(q.x, "\n")
	fmt.Print(q.y, "\n")
	fmt.Print(q.z, "\n")
}
