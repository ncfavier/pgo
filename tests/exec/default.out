false
0

0
false
<nil>
<nil>
0
false
<nil>
